internal/client.go

```diff
@@ -50,8 +50,32 @@ type (
 		//	- EntityNotExistsError
 		//	- BadRequestError
 		//	- WorkflowExecutionAlreadyStartedError
+		//	- InternalServiceError
 		StartWorkflow(ctx context.Context, options StartWorkflowOptions, workflow interface{}, args ...interface{}) (*WorkflowExecution, error)
 
+		// ExecuteWorkflow starts a workflow execution and return a WorkflowRun instance and error
+		// The user can use this to start using a function or workflow type name.
+		// Either by
+		//     ExecuteWorkflow(ctx, options, "workflowTypeName", input)
+		//     or
+		//     ExecuteWorkflow(ctx, options, workflowExecuteFn, arg1, arg2, arg3)
+		// The errors it can return:
+		//	- EntityNotExistsError
+		//	- BadRequestError
+		//	- WorkflowExecutionAlreadyStartedError
+		//	- InternalServiceError
+		//
+		// WorkflowRun has 2 methods:
+		//  - GetRunID() string: which return the first started workflow run ID (please see below)
+		//  - Get(ctx context.Context, valuePtr interface{}) error: which will fill the workflow
+		//    execution result to valuePtr, if workflow execution is a success, or return corresponding
+		//    error. This is a blocking API.
+		// NOTE: if the started workflow return ContinueAsNewError during the workflow execution, the
+		// return result of GetRunID() will NOT be updated, however, Get(ctx context.Context, valuePtr interface{})
+		// will use the new run ID for workflow execution result.
+		// NOTE: DO NOT USE THIS API INSIDE A WORKFLOW
+		ExecuteWorkflow(ctx context.Context, options StartWorkflowOptions, workflow interface{}, args ...interface{}) (WorkflowRun, error)
```

Comment:
I thought we discussed and agreed on using RunWorkflow as method name and returning an encoded.Value and error. Looks like I missed some context here?

Reply:
mfateev, samarabbas and wxing1292 had a discussion on last Friday, the idea is to make the ExecuteWorkflow (renamed) API having the same user experience as the ExecuteChildWorkflow

Reply:
@yiminc FYI, i am adding unit test